Source/CarthageKit/DiagnosticResolver.swift

```diff
@@ -0,0 +1,170 @@
+import Foundation
+import Result
+import ReactiveSwift
+
+/**
+Delegate for listening to resolver events.
+*/
+public protocol DiagnosticResolverDelegate: class {
+	func diagnosticResolver(_ resolver: DiagnosticResolver, didFindVersions versions: [PinnedVersion], compatibleWith versionSpecifier: VersionSpecifier, for dependency: Dependency)
+	func diagnosticResolver(_ resolver: DiagnosticResolver, didFindTransitiveDependencies transitiveDependencies: [(Dependency, VersionSpecifier)], for dependency: Dependency, withVersion version: PinnedVersion)
+	func diagnosticResolver(_ resolver: DiagnosticResolver, didEncounterError error: Error, whileRetrievingTransitiveDependenciesFor dependency: Dependency, withVersion version: PinnedVersion)
+	func diagnosticResolver(_ resolver: DiagnosticResolver, didEncounterError error: Error, whileRetrievingVersionsFor dependency: Dependency, compatibleWith versionSpecifier: VersionSpecifier)
+}
+
+/**
+Resolver which logs all dependencies it encounters, and optionally stores them in the specified local repository.
+*/
+public final class DiagnosticResolver: ResolverProtocol {
+	public weak var delegate: DiagnosticResolverDelegate?
```

Comment:
I don't think Carthage uses `delegate` pattern. Please consider to use `Signal` instead.

Reply:
I agree. You can find several examples of how to use Signal for this purpose in the existing code base

Reply:
I don't understand how to apply this without changing the resolver protocol. Please either make the change yourself or provide a more elaborate code example.